File: python/0169-majority-element.py
class Solution:
    # O(n) time | O(1) space
    def majorityElement(self, nums: List[int]) -> int:
        val, count = nums[0], 0
        for num in nums:
            if num == val:
                count += 1
            else:
                count -= 1
                if count == 0:
                    val = num
                    count = 1
        return val

File: c++/0169-majority-element.cpp
class Solution {
public:
    // O(n) time, O(1) space - where n is the size of the input array
    int majorityElement(vector<int>& nums) {
        int res = nums[0], count = 1;
        int n = nums.size();
        for (int i = 1; i < n; ++i) {
            if (nums[i] != res) {
                --count;
            }
            if (count == 0) {
                res = nums[i];
                count = 1;
            }
        }
        return res;
    }
};